numpy/core/src/umath/scalarmath.c.src

```diff
@@ -1058,85 +976,25 @@ static PyObject *
         return Py_NotImplemented;
     }
 
+#if !@isint@
     npy_clear_floatstatus_barrier((char*)&out);
-
+#endif
     /*
      * here we do the actual calculation with arg1 and arg2
      * as a function call.
      */
+#if @isint@ && !@isuint@
```

Comment:
These are already mutually exclusive, right? - just `@isint@` would be enough

Reply:
ops, I changed it the opposite direction, isint includes uint. The reason is that the float checking is not necessary for all ints here.

Added one more commit, before I forget it, although its unrelated. It just deletes a `volatile` decleration with the comment, since the code now uses `npy_clear_floatstatus_barrier` which solves the same issue as declaring `out` as `volatile`.